src/Xamarin.Android.Build.Tasks/Utilities/JCWGenerator.cs

```diff
@@ -220,7 +221,7 @@ static string GetMonoInitSource (string androidSdkPlatform)
 		return builder.ToString ();
 	}
 
-	public static void EnsureAllArchitecturesAreIdentical (TaskLoggingHelper logger, Dictionary<AndroidTargetArch, NativeCodeGenState> javaStubStates)
+	public static void EnsureAllArchitecturesAreIdentical (TaskLoggingHelper logger, ConcurrentDictionary<AndroidTargetArch, NativeCodeGenState> javaStubStates)
```

Comment:
Did you test performance with a `lock` instead of using `ConcurrentDictionary`? A monitor might perform better than any concurrent collection.

Reply:
I did not. We only store a maximum of 4 items in the `Dictionary`, so I'm not sure any operations on it will be measurable.